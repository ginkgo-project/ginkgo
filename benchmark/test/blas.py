#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"n": 100}]'],
    expected_stdout="blas.simple.stdout",
    expected_stderr="blas.simple.stderr",
)

# stdin
test_framework.compare_output(
    [],
    expected_stdout="blas.simple.stdout",
    expected_stderr="blas.simple.stderr",
    stdin='[{"n": 100}]',
)

# file
test_framework.compare_output(
    ["-input", str(test_framework.sourcepath / "input.blas.json")],
    expected_stdout="blas.simple.stdout",
    expected_stderr="blas.simple.stderr",
    stdin='[{"n": 100}]',
)

# profiler annotations
test_framework.compare_output(
    ["-input", '[{"n": 100}]', "-profile", "-profiler_hook", "debug"],
    expected_stdout="blas.profile.stdout",
    expected_stderr="blas.profile.stderr",
    stdin='[{"n": 100}]',
)
