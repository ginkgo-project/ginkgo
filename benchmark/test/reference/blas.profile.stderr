Running on reference(0)
Running with 0 warm iterations and 1 running iterations
The random seed for right hand sides is 42
The operations are copy,axpy,scal
Running test case n = 100 
DEBUG: begin n = 100 
	Running blas: copy
DEBUG: begin copy
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin repetition
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   repetition
DEBUG: end   copy
	Running blas: axpy
DEBUG: begin axpy
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin repetition
DEBUG: begin dense::add_scaled
DEBUG: end   dense::add_scaled
DEBUG: end   repetition
DEBUG: end   axpy
	Running blas: scal
DEBUG: begin scal
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin repetition
DEBUG: begin dense::scale
DEBUG: end   dense::scale
DEBUG: end   repetition
DEBUG: end   scal
DEBUG: end   n = 100 
